minio/api.py

```diff
@@ -578,6 +578,73 @@ def remove_all_bucket_notification(self, bucket_name):
             content_sha256=content_sha256_hex
         )
 
+    # put_bucket_encryption sets default encryption configuration on an existing bucket.
+    def put_bucket_encryption(self, bucket_name, enc_config):
+        """
+        Set default encryption configuration on a given bucket.
+
+        :param bucket_name: Bucket name.
+        :param enc_config: Default encryption configuration in dictionary format.
+        """
+        is_valid_bucket_name(bucket_name, False)
+
+        apply_enc = enc_config["ServerSideEncryptionConfiguration"]["Rule"][0]["ApplyServerSideEncryptionByDefault"]
+        sse_value = apply_enc["SSEAlgorithm"]
+        if 'KMSMasterKeyID' in apply_enc.keys():
+            kms_value = apply_enc["KMSMasterKeyID"]
+        else:
+            kms_value = ''
+        enc_config_xml = xml_marshal_bucket_encryption(sse_value, kms_value)
+
+        headers = {
+            'Content-Length': str(len(enc_config_xml)),
+            'Content-Md5': get_md5_base64digest(enc_config_xml)
+        }
+        content_sha256_hex = get_sha256_hexdigest(enc_config_xml)
+        self._url_open("PUT",
+                        bucket_name=bucket_name,
+                        query={"encryption": ""},
+                        headers=headers,
+                        body=enc_config_xml,
+                        content_sha256=content_sha256_hex
+                        )
+
+    def get_bucket_encryption(self, bucket_name):
+        """
+        Get default encryption configuration information on a given bucket.
+
+        :param bucket_name: Bucket name.
+        """
+        result = ''
+        is_valid_bucket_name(bucket_name, False)
+        try:
+            response = self._url_open(
+                "GET",
+                bucket_name=bucket_name,
+                query={"encryption": ""}
+            )
+            result = response.data.decode('utf-8')
+        except ResponseError as err:
```

Comment:
I think we should not catch any exception here

Reply:
Sure, I'll remove it.